### PR TITLE
Add a clean function for spack-manager

### PR DESCRIPTION
**Pull request by psakievich**
Tribal knowledge function for cleaning things up: `sm-clean`. Let me know if I'm missing anything. Pinging @PaulMullowney as well.

Also adding some missing help messages from functions.

### CODE REVIEW

File: docs/general/FAQ.md [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 
 ## I'm getting lots of weird Spack errors, what should I do?
-If you have another Spack instance you've used in the past conflicts can occur due to the `~/.spack` directory.
-The first step we recommend is deleting this directory and trying again.
+If you have multiple spack instances or have recently updated spack things can get a little mangled.
```
Comment (tasmith4) [inline docs/general/FAQ.md]: I think it’s still worth mentioning one reason is that both spack instances store things in ~/.spack

File: start.sh [hunk 2]
```diff
@@ -172,3 +186,17 @@ function build-env-dive() {
...
+
+# function for cleaning spack-manager and spack directories
+function sm-clean(){
+  if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+    echo "This command will clean out spack and spack-manager caches and other problematic directories"
+    return
+  fi
+  echo 'Remove user cache of configs:'
+  cmd "rm -rf ~/.spack"
+  echo 'Use spack's native clean system:
```
Comment (tasmith4) [inline start.sh]: Probable typo, missing single quote (actually worse than that, the single quote in “spack’s” closes the one at the beginning of the command)

**Review by tasmith4 — COMMENTED** [review]
I wonder if this file is getting big enough that it’s worth consolidating in some way or doing it in Python

File: docs/general/FAQ.md (hunk 1 above)
Comment (psakievich) [reply docs/general/FAQ.md]: Sure I can add that

**Comment by psakievich** [comment]
I thought about consolidating or going to python.  The main thing with these functions is we want them to execute in the active shell.  `sm-clean` could probably be python but all the rest probably not.  I'd rather leave it as is for now, but I think moving things around for better organization is on the table for the future.

**Review by jrood-nrel — APPROVED** [review]

**Review by tasmith4 — APPROVED** [review]

**Pull request merged**